protocol/communities/persistence.go

```diff
@@ -161,6 +149,38 @@ WHERE c.Joined OR r.state = ?`
 	return comms, nil
 }
 
+func (p *Persistence) JoinedAndPendingCommunitiesWithRequests(memberIdentity *ecdsa.PublicKey) (comms []*Community, err error) {
+	query := `SELECT
+c.id, c.private_key, c.description, c.joined, c.verified, c.muted,
+r.id, r.public_key, r.clock, r.ens_name, r.chat_id, r.community_id, r.state
+FROM communities_communities c
+LEFT JOIN communities_requests_to_join r ON c.id = r.community_id AND r.public_key = ?
+WHERE c.Joined OR r.state = ?`
+
+	rows, err := p.db.Query(query, common.PubkeyToHex(memberIdentity), RequestToJoinStatePending)
+	if err != nil {
+		return nil, err
+	}
+
+	return p.rowsToCommunities(memberIdentity, rows)
+}
+
+func (p *Persistence) DeletedCommunities(memberIdentity *ecdsa.PublicKey) (comms []*Community, err error) {
+	query := `SELECT
+c.id, c.private_key, c.description, c.joined, c.verified, c.muted,
+r.id, r.public_key, r.clock, r.ens_name, r.chat_id, r.community_id, r.state
+FROM communities_communities c
+LEFT JOIN communities_requests_to_join r ON c.id = r.community_id AND r.public_key = ?
+WHERE NOT c.Joined AND (r.community_id IS NULL or r.state != ?)`
```

Comment:
Hey @rasom , I am trying to understand why `r.community_id IS NULL` condition is needed, do you remember by any chance? At first glance it doesn't make much sense.

CC: @kounkou

Reply:
He's on parental leave

Reply:
Congrats!! :baby:  And sorry for bothering.